darwin/importer/importer.py

```diff
@@ -267,6 +274,159 @@ def _resolve_annotation_classes(
     return local_classes_not_in_dataset, local_classes_not_in_team
 
 
+def _import_properties(
+    metadata_path: Union[Path, bool],
+    client: "Client",
+    annotations: List[dt.Annotation],
+    annotation_class_ids_map: Dict[str, str],
+):
+    if not isinstance(metadata_path, Path):
+        # No properties to import
+        return
+
+    # parse metadata.json file -> list[PropertyClass]
+    metadata = parse_metadata(metadata_path)
+    metadata_property_classes = parse_property_classes(metadata)
+
+    # get team properties -> List[FullProperty]
+    team_properties = client.get_team_properties()
+    # (property-name, annotation_class_id): FullProperty object
+    team_properties_annotation_lookup: Dict[
+        Tuple[str, Optional[int]], FullProperty
+    ] = {}
+    for prop in team_properties:
+        team_properties_annotation_lookup[(prop.name, prop.annotation_class_id)] = prop
+
+    # (annotation-cls-name, annotation-cls-name): PropertyClass object
+    metadata_classes_lookup: List[Tuple[str, str]] = []
+    # (annotation-cls-name, property-name): Property object
+    metadata_cls_prop_lookup: Dict[Tuple[str, str], Property] = {}
+    for _cls in metadata_property_classes:
+        metadata_classes_lookup += [(_cls.name, _cls.type)]
+        for _prop in _cls.properties or []:
+            metadata_cls_prop_lookup[(_cls.name, _prop.name)] = _prop
+
+    create_properties: List[FullProperty] = []
+    update_properties: List[FullProperty] = []
+    for annotation in annotations:
+        annotation_name = annotation.annotation_class.name
+        annotation_type = annotation_type = (
+            annotation.annotation_class.annotation_internal_type
+            or annotation.annotation_class.annotation_type
+        )
+        annotation_class_id = int(annotation_class_ids_map[annotation_name])
```

Comment:
I think this method might have an issue with conflicting annotation_names, which can happen. It's possible to have both a polygon and a line annotation be the same name. Only (annotation_name, annotation_type/class) is unique.

Reply:
Good catch! Updated the variable to have `(annotation_class.name, annotation_type)` as keys now!